## Assignment 4A

### Problem 1 (50 points)
Wien’s displacement constant. Planck’s radiation law tells us that the intensity of radiation per unit area and per unit wavelength $\lambda$ from a black body at temperature $T$ is:
$$I(\lambda)=\dfrac{2\pi hc^2}{\lambda^5}\dfrac{1}{e^{hc/\lambda k_BT}-1}$$
where $h$ is Planck’s constant, $c$ is the speed of light, and $k_B$ is Boltzmann’s constant.

---

#### Problem 1.A (10 points)
Show by differentiating that the wavelength $\lambda$ at which the emitted radiation is strongest is the solution of the equation:
$$5e^{-hc/\lambda k_BT}+\dfrac{hc}{\lambda k_BT}-5=0$$
Make the substitution $x=\dfrac{hc}{\lambda k_BT}$ and hence show that the wavelength of maximum radiation obeys the Wien displacement law:
$$\lambda=\dfrac{b}{T}$$
where the so-called Wien displacement constant is
$$b=\dfrac{hc}{k_Bx}$$
and x is the solution to the nonlinear equation
$$5e^{-x}+x-5$$

$$\begin{align}
I(\lambda)&=2\pi hc^2\lambda^{-5}\left(e^{hc/\lambda k_BT}-1\right)^{-1}\\
I_\lambda(\lambda)&=2\pi hc^2\cdot\dfrac{d}{d\lambda}\left(\lambda^{-5}\left(e^{hc/\lambda k_BT}-1\right)^{-1}\right)\\
&=2\pi hc^2\cdot\dfrac{d}{d\lambda}\left(\lambda^{-5}\right)\cdot\left(e^{hc/\lambda k_BT}-1\right)^{-1}+2\pi hc^2\cdot\lambda^{-5}\cdot\dfrac{d}{d\lambda}\left(e^{hc/\lambda k_BT}-1\right)^{-1}\\
&=2\pi hc^2\cdot\left(-\dfrac{5}{\lambda}\right)\left(\lambda^{-5}\cdot\left(e^{hc/\lambda k_BT}-1\right)^{-1}\right)+2\pi hc^2\cdot\lambda^{-5}\cdot(-1)\left(e^{hc/\lambda k_BT}-1\right)^{-2}\dfrac{d}{d\lambda}\left(e^{hc/\lambda k_BT}-1\right)\\
0&=\left(-\dfrac{5}{\lambda}\right)I(\lambda)+(-1)\left(e^{hc/\lambda k_BT}-1\right)^{-1}\dfrac{d}{d\lambda}\left(e^{hc/\lambda k_BT}-1\right)I(\lambda)\\
0&=\left(\dfrac{5}{\lambda}\right)+\left(e^{hc/\lambda k_BT}-1\right)^{-1}\left(e^{hc/\lambda k_BT}\right)\dfrac{d}{d\lambda}\left( \dfrac{hc}{k_BT}\dfrac{1}{\lambda}\right)\\
0&=\left(\dfrac{5}{\lambda}\right)-\left(e^{hc/\lambda k_BT}-1\right)^{-1}\left(e^{hc/\lambda k_BT}\right)\left( \dfrac{hc}{k_BT}\dfrac{1}{\lambda^2}\right)\\
0&=5\left(e^{hc/\lambda k_BT}-1\right)-\left(e^{hc/\lambda k_BT}\right)\left( \dfrac{hc}{\lambda k_BT}\right)\\
0&=5e^{hc/\lambda k_BT}-5-\left(e^{hc/\lambda k_BT}\right)\left( \dfrac{hc}{\lambda k_BT}\right)\\
0&=5-5e^{-hc/\lambda k_BT}-\left( \dfrac{hc}{\lambda k_BT}\right)\\
0&=5e^{-hc/\lambda k_BT}+\dfrac{hc}{\lambda k_BT}-5
\end{align}$$
---

#### Problem 1.B (35 points)
Write a program to solve this equation to an accuracy of $10^{−6}$ using the binary search method, and hence find a value for the displacement constant.

In [3]:
import numpy as np

# Binary partition method
def bisection_method(f: callable, a: int | float, b: int | float, *, error: int | float, max_iter: int) -> float:
    assert callable(f)
    assert isinstance(a, (int, float))
    assert isinstance(b, (int, float))
    assert isinstance(error, (int, float))
    assert isinstance(max_iter, int)
    assert a < b
    assert f(a) * f(b) < 0
    for _ in range(max_iter):
        c = (a + b) / 2
        if np.abs(f(c)) < error:
            print(f"Error is less than {error}")
            print(f"Number of iterations: {_}")
            return c
        if f(a) * f(c) < 0:
            b = c
        else:
            a = c
    print(f"Error is greater than {error}")
    print(f"Number of iterations: {max_iter}")
    return c

# Problem data
f: callable = lambda x: 5*np.exp(-x) + x - 5
A = 4
B = 6
error = 1e-6
max_iter = 10_000

# Solve the problem
val = bisection_method(f, A, B, error=error, max_iter=max_iter)

# Output
print(f"The solution is: {val}")

Error is less than 1e-06
Number of iterations: 18
The solution is: 4.965114593505859


The solution I got was $4.965114593505859$ from the equation given above.

---

#### Problem 1.C (5 points)
The displacement law is the basis for the method of optical pyrometry, a method for measuring the temperatures of objects by observing the color of the thermal radiation they emit. The method is commonly used to estimate the surface temperatures of astronomical bodies, such as the Sun. The wavelength peak in the Sun’s emitted radiation falls at $\lambda=502\text{nm}$. From the equations above and your value of the displacement constant, estimate the surface temperature of the Sun.
$$\begin{align}
x&=4.965114593505859\\
b&=\dfrac{hc}{k_B}\dfrac{1}{x}\\
&=0.01439\text{ m}\cdot\text{K}\dfrac{1}{4.965\dots}\\
b&=0.00289784\text{ m}\cdot\text{K}
\end{align}$$
Using this value of the Wien displacement constant, we can see that:
$$\begin{align}
&&\lambda &= 502\cdot10^{-9}\text{ m}\\
&&b&=0.00289784\text{ m}\cdot\text{K}\\
\implies&&T&=\dfrac{0.00289784\text{ m}\cdot\text{K}}{502\cdot10^{-9}\text{ m}}\\
&&T&=5772.58964143\text{ K}
\end{align}$$
---

### Problem 2 (50 points)
The Lagrange point. There is a special point between the Earth and the Moon, called the L1 Lagrange point, at which a satellite will orbit the Earth in perfect synchrony with the Moon, staying always in between the two. This works because the inward pull of the Earth and the outward pull of the Moon combine to create exactly the needed centripetal force that keeps the satellite in its orbit.

Assuming circular orbits, and assuming that the Earth is much more massive than either the Moon or the satellite, show that the distance r from the center of the Earth to the L1 point satisfies:
$$\dfrac{GM}{r^2}-\dfrac{Gm}{(R-r)^2}=\omega^2 r$$
where $M$ and $m$ are the Earth and Moon masses, $G$ is Newton’s gravitational constant, and $\omega$ is the angular velocity of both the Moon and the satellite.

The equation above is a fifth-order polynomial equation in $r$ (also called a quintic equation). Such equations cannot be solved exactly in closed form, but it’s straightforward to solve them numerically. Write a program that uses the secant method to solve for the distance $r$ from the Earth to the $L_1$ point. Compute a solution accurate to at least four significant digits. The values of the various parameters are:
$$\begin{align}
G&=6.674\cdot10^{-11}\text{ m}^3\text{ kg}^{-1}\text{s}^{-2}\\
M&=5.974\cdot10^{24}\text{ kg}\\
m&=7.348\cdot10^{22}\text{ kg}\\
R&=3.844\cdot10^{8}\text{ m}\\
\omega&=2.662\cdot10^{-6}\text{ s}^{-1}
\end{align}$$

In [5]:
import numpy as np

# Function Definitoon
def secant_method(*, f: callable, x0: int | float, x1: int | float, error: int | float, max_iter: int) -> float:
    assert callable(f)
    assert isinstance(x0, (int, float))
    assert isinstance(x1, (int, float))
    assert isinstance(error, (int, float))
    assert isinstance(max_iter, int)
    assert error > 0
    assert max_iter > 0
    for _ in range(max_iter):
        x2 = x1 - f(x1) * (x1 - x0) / (f(x1) - f(x0))
        if np.abs(x2 - x1) < error:
            print(f"Error is less than {error}")
            print(f"Number of iterations: {_}")
            return x2
        x0, x1 = x1, x2
    print(f"Error is greater than {error}")
    print(f"Number of iterations: {max_iter}")
    return x2

# Problem data
G = 6.674e-11  # Gravitational constant
M = 5.972e24   # Earth mass
m = 7.348e22   # Moon mass
R = 3.844e8    # Distance between Earth and Moon
w = 2.662e-6   # Angular velocity of the Moon

# Function definition
f: callable = lambda r: G * M / r**2 - G * m / (R - r)**2 - r * w**2

# Initial guess
X0 = 1e7
X1 = 1.5e7

# Solve the problem
val = secant_method(f=f, x0=X0, x1=X1, error=1e-6, max_iter=10_000)
print(f"The solution is: r={val}")

Error is less than 1e-06
Number of iterations: 17
The solution is: r=326029262.49331266


From my calculation, I found that the object would need to be $3.2603\cdot10^8\text{ m}$ away from Earth in order to stay in-place.